lib/mshoplib/src/MShop/Index/Manager/Price/Standard.php

```diff
@@ -653,11 +653,26 @@ protected function savePrices( \Aimeos\MW\DB\Statement\Iface $stmt, \Aimeos\MSho
 		$context = $this->getContext();
 		$siteid = $context->getLocale()->getSiteId();
 
-		foreach( $item->getListItems( 'price', 'default', 'default' ) as $listItem )
+		/** mshop/index/manager/price/price-types
+		 * Use different product prices types for sorting by price
+		 *
+		 * In some cases, prices are stored with different types, eg. price per kg.
+		 * This configuration option defines which types are incorporated when sorting
+		 * the product list by price.
+		 *
+		 * @param array List of price types codes
+		 * @since 2019.1
+		 * @category Developer
+		 */
+		$types = $context->getConfig()->get( 'mshop/index/manager/price/price-types', ['default'] );
+		foreach( $item->getListItems( 'price', 'default', $types ) as $listItem )
 		{
-			if( ( $refItem = $listItem->getRefItem() ) !== null && $refItem->isAvailable() ) {
-				$prices[$refItem->getCurrencyId()][$refItem->getQuantity()] = $refItem->getValue();
-			}
+			if( ( $refItem = $listItem->getRefItem() ) !== null 
+				&& ( !isset ( $prices[$refItem->getCurrencyId()][$refItem->getQuantity()] ) 
+				|| $prices[$refItem->getCurrencyId()][$refItem->getQuantity()] > $refItem->getValue() ) )
```

Comment:
This line has no effect because there aren't several prices for the same quantity (usually, if customer specific prices are ignored). Furthermore, you've removed the `$refItem->isAvailable()` condition which is necessary to skip prices that are not yet available or not available any more.